context/context.go

```diff
@@ -207,7 +207,11 @@ func (c *fsContext) Branch() (string, error) {
 	}
 
 	currentBranch, err := git.CurrentBranch()
-	if err != nil {
+	switch err {
```

Comment:
I think I've addressed this concern to a degree.

- update the "blank context" mock to return an `ErrNotOnAnyBranch` if no branch has been set.
- add a test to `pr_test.go` that checks that the command runs to completion if there is no branch.

I don't think this is completely foolproof, since it relies on the mock's behavior staying synchronized with the regular filesystem context, so I'm not 100% happy with it.

But I didn't see any existing tests that set up a real Git repository (on a real filesystem, using real `git`), and I'm assuming that you don't want to add that with this change.

Reply:
I was imagining using our `CmdStubber`'s `StubError` to have the `git branch` call result in the expected error. Sorry for the back and forth but I'd rather see that approach (unless there's a reason why it wouldn't work that I'm forgetting about).